barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_trace.cpp

```diff
@@ -3766,7 +3766,9 @@ std::vector<Row> AvmTraceBuilder::finalize(uint32_t min_trace_size, bool range_c
     // If the bin_trace_size has entries, we need the main_trace to be as big as our byte lookup table (3 *
     // 2**16 long)
     size_t const lookup_table_size = (bin_trace_size > 0 && range_check_required) ? 3 * (1 << 16) : 0;
-    size_t const range_check_size = range_check_required ? UINT16_MAX + 1 : 0;
+    // Range check size is 1 less than it needs to be since we insert a "first row" at the top of the trace at the end,
+    // with clk 0 (this doubles as our range check)
+    size_t const range_check_size = range_check_required ? UINT16_MAX : 0;
```

Comment:
@IlyasRidhuan Isn't it the same for lookup_table_size ? Any other "trace_size" that we should adapt?

Reply:
Yeah it's true for other lookups like the binary trace, I've left that for now since we will do a larger refactor to finalize.

Plus it's less of a concern because they do not create a table whose rows are near boundary of a power of 2 (so we dont gain any performance benefits).